packages/mui-system/src/sx/sx.js

```diff
@@ -0,0 +1,7 @@
+import styleFunctionSx from '../styleFunctionSx';
+
+function sx(styles, theme) {
+  return styleFunctionSx({ sx: styles, theme });
+}
```

Comment:
Can you try with:
```js
const sx = (styles) => ({ theme }) => styleFunctionSx({ sx: styles, theme });
```

From what I check in emotion & styled-components, they support nested function. So we don't need to get theme if we don't need it.

```js
const MyThemeComponent = styled('div')(
  sx({
      color: 'primary.contrastText',
      backgroundColor: 'primary.main',
      padding: 1,
      borderRadius: 1,
    })
);
```

Reply:
I was trying to create an API not tightly related to the `styled()`, but maybe this is the only use-case, so yeah, let me update

Reply:
Updated. I've tested it using in the `styled()` utility and in the variants in side the theme. I can't think of any other use-cases at this moment.